### PR TITLE
documented windows filters

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

filters

### CODE REVIEW

File: lib/ansible/plugins/filter/win_basename.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+DOCUMENTATION:
+  name: win_basename
+  author: ansible core team
+  version_added: "2.0"
+  short_description: get a Windows path's base name
+  description:
+    - Returns the last name component of a Windows path, what is left in the string that is not 'win_dirname'.
+  options:
+    _input:
+      description: A Windows path
```
Comment (felixfontein) [inline lib/ansible/plugins/filter/win_basename.yml]:
```suggestion
      description: A Windows path.
```

File: lib/ansible/plugins/filter/win_dirname.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+DOCUMENTATION:
+  name: win_dirname
+  author: ansible core team
+  version_added: "2.0"
+  short_description: get a Windows path's directory
+  description:
+    - Returns the directory component of a Windows path, what is left in the string that is not 'win_basename'.
+  options:
+    _input:
+      description: A Windows path
```
Comment (felixfontein) [inline lib/ansible/plugins/filter/win_dirname.yml]:
```suggestion
      description: A Windows path.
```

File: lib/ansible/plugins/filter/win_splitdrive.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+DOCUMENTATION:
+  name: win_splitdrive
+  author: ansible core team
+  version_added: "2.0"
+  short_description: split a Windows path by the drive letter
+  description:
+    - Returns a list with the first component being the drive letter and the second, the rest of the path.
+  options:
+    _input:
+      description: A Windows path
```
Comment (felixfontein) [inline lib/ansible/plugins/filter/win_splitdrive.yml]:
```suggestion
      description: A Windows path.
```

File: lib/ansible/plugins/filter/win_splitdrive.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  # just the drive letter
+  {{ mypath | win_splitdrive|first }}
+
+  # path w/o drive letter
+  {{ mypath | win_splitdrive|last }}
+
+RETURN:
+  _value:
+    description: List in which the first element is the drive letter and the 2nd the rest of the path
```
Comment (felixfontein) [inline lib/ansible/plugins/filter/win_splitdrive.yml]:
```suggestion
    description: List in which the first element is the drive letter and the second the rest of the path.
```

File: lib/ansible/plugins/filter/win_splitdrive.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+EXAMPLES: |
+
+  # To get the last name of a file Windows path, like ['C', '\Users\asdf\foo.txt'] out of 'C:\Users\asdf\foo.txt'
+  {{ mypath | win_splitdrive }}
+
+  # just the drive letter
+  {{ mypath | win_splitdrive|first }}
+
+  # path w/o drive letter
+  {{ mypath | win_splitdrive|last }}
```
Comment (felixfontein) [inline lib/ansible/plugins/filter/win_splitdrive.yml]:
```suggestion
  {{ mypath | win_splitdrive | first }}

  # path w/o drive letter
  {{ mypath | win_splitdrive | last }}
```

File: lib/ansible/plugins/filter/win_dirname.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    - plugin_type: filter
+      plugin: ansible.builtin.win_basename
+EXAMPLES: |
+
+  # To get the last name of a file Windows path, like 'C:\users\asdf' out of 'C:\Users\asdf\foo.txt'
+  {{ mypath | win_dirname}}
+
+RETURN:
+  _value:
+    description: The direcotry from the Windows path provided.
```
Comment (felixfontein) [inline lib/ansible/plugins/filter/win_dirname.yml]:
```suggestion

RETURN:
  _value:
    description: The directory from the Windows path provided.
```

File: lib/ansible/plugins/filter/win_dirname.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      description: A Windows path
+      type: str
+      required: true
+  seealso:
+    - plugin_type: filter
+      plugin: ansible.builtin.win_basename
+EXAMPLES: |
+
+  # To get the last name of a file Windows path, like 'C:\users\asdf' out of 'C:\Users\asdf\foo.txt'
+  {{ mypath | win_dirname}}
```
Comment (felixfontein) [inline lib/ansible/plugins/filter/win_dirname.yml]:
```suggestion
  {{ mypath | win_dirname }}
```

**Comment by felixfontein** [comment]
A rendered version of the current PR's state can be seen here:
- https://ansible.fontein.de/collections/ansible/builtin/win_basename_filter.html#ansible-collections-ansible-builtin-win-basename-filter
- https://ansible.fontein.de/collections/ansible/builtin/win_dirname_filter.html#ansible-collections-ansible-builtin-win-dirname-filter
- https://ansible.fontein.de/collections/ansible/builtin/win_splitdrive_filter.html#ansible-collections-ansible-builtin-win-splitdrive-filter

**Review by felixfontein — APPROVED** [review]

File: lib/ansible/plugins/filter/win_basename.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+DOCUMENTATION:
+  name: win_basename
+  author: ansible core team
+  version_added: "2.0"
+  short_description: get a Windows path's base name
```
Comment (samccann) [inline lib/ansible/plugins/filter/win_basename.yml]:
```suggestion
  short_description: Get a Windows path's base name
```

File: lib/ansible/plugins/filter/win_dirname.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+DOCUMENTATION:
+  name: win_dirname
+  author: ansible core team
+  version_added: "2.0"
+  short_description: get a Windows path's directory
```
Comment (samccann) [inline lib/ansible/plugins/filter/win_dirname.yml]:
```suggestion
  short_description: Get a Windows path's directory
```

File: lib/ansible/plugins/filter/win_splitdrive.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
+DOCUMENTATION:
+  name: win_splitdrive
+  author: ansible core team
+  version_added: "2.0"
+  short_description: split a Windows path by the drive letter
```
Comment (samccann) [inline lib/ansible/plugins/filter/win_splitdrive.yml]:
```suggestion
  short_description: Split a Windows path by the drive letter
```

**Review by samccann — APPROVED** [review]

**Pull request merged**